### DISCUSSION TITLE
pivot point -  how to set center of rotation

### DISCUSSION THREAD

**Discussion by itakka01** [mkkellogg/GaussianSplats3D#210] [General]
Thank you very much for providing this excellent program. 
I'm curious if there's a way to adjust the pivot/center of rotation. Despite setting it to the center of the object using my standard editor  supersplat, I'm finding it difficult to rotate it properly ( example:  https://t4.itakka.at/3dstuff/kapelle1 ). 

Is there a feature/setting to  adjust the pivot to the center of the object before the viewer starts, or could you recommend a program that offers this capability?

**Comment by mkkellogg** [comment]
Sure, you can use the `initialCameraLookAt` constructor parameter for the `Viewer` to set the initial center of rotation. You can also use the `cameraUp` parameter to set the axis of rotation:
```
const viewer = new GaussianSplats3D.Viewer({
    'cameraUp': [0.02356, -0.99964, 0.01263],
    'initialCameraLookAt': [0.09114, -0.13449, 0.23314]
});
```

If you're having trouble determining the best `cameraUp` and `initialCameraLookAt`,  there are built-in tools to help you. You can adjust the camera's up vector in real-time using the left or right arrow keys. You can also press U to view a debug object that shows the orientation of the camera's orbital plane. That may help you find the camera up vector that works for your scene. For the scene you're testing with, [0.02356, -0.99964, 0.01263] seems to work well for `cameraUp`, along with [0.09114, -0.13449, 0.23314] for `initialCameraLookAt`.
